python_modules/libraries/dagster-airbyte/dagster_airbyte/managed/reconciliation.py

```diff
@@ -587,3 +598,74 @@ def apply(self) -> ManagedElementCheckResult:
             dry_run=False,
             should_delete=self._delete_unmentioned_resources,
         )
+
+
+class AirbyteManagedElementCacheableAssetsDefinition(AirbyteInstanceCacheableAssetsDefintion):
+    def __init__(
+        self,
+        airbyte_resource_def: ResourceDefinition,
+        key_prefix: List[str],
+        create_assets_for_normalization_tables: bool,
+        connection_to_group_fn: Optional[Callable[[str], Optional[str]]],
+        connections: Iterable[AirbyteConnection],
+    ):
+        super().__init__(
+            airbyte_resource_def=airbyte_resource_def,
+            workspace_id=None,
+            key_prefix=key_prefix,
+            create_assets_for_normalization_tables=create_assets_for_normalization_tables,
+            connection_to_group_fn=connection_to_group_fn,
+            connection_filter=None,
+        )
+        self._connections: List[AirbyteConnection] = list(connections)
+
+    def _get_connections(self) -> List[Tuple[str, AirbyteConnectionMetadata]]:
+        diff = reconcile_config(self._airbyte_instance, self._connections, dry_run=True)
```

Comment:
what will happen when airbyte instance is down? does this mean when the airbyte instance is down, the entire repo will error?

Reply:
Yes, maybe a user option to not load these assets? Not sure. We can’t fully construct them without access to the instance

Reply:
I think failing to load the repo when you can't access the airbyte instance is generally the correct / desirable behavior. This will only happen on repository reload (generally users in prod won't be clicking this button themselves, the reload will mostly just be triggered by merges into the main branch), and if you can't access the airbyte instance at that time, it'd be pretty bad to silently drop the airbyte assets. asset jobs defined with stuff like AssetSelection.upstream("my_cool_asset") would suddenly no longer include airbyte-related stuff, among other similar issues.

The specific order of events which is maximally disruptive is:

1) user's airbyte service goes down in a pretty bad way (will take active work to fix, rather than just waiting a minute)
2) a PR is merged (or someone manually clicks the reload definitions button), causing the repository definition to be reloaded. this causes an error, as we can't reload this definition from scratch w/o access to the instance
3) while the airbyte issue is being resolved, we want to run jobs that are not related to airbyte

The remedy for this would be for the user to actively remove the airbyte assets from their repo (i.e. pushing a code change) until the root airbyte issue is solved. This is annoying, but I think preferable to silently doing a similar thing on the user's behalf.